main.py

```diff
@@ -0,0 +1,423 @@
+
+    if (existing_truck := collection.find_one({"_id": ObjectId(truck_id)})) is not None:
+        return serialize_collection(existing_truck)
+
+@app.put("/trucks/{truck_id}/add_repair", response_model=Truck)
+async def update_repairs(truck_id: str, repair: RepairModel = Body(...)):
+
+    collection = db.get_collection("trucks")
+
+    truck = collection.find_one({"_id": ObjectId(truck_id)})
+    if truck is None:
+        raise HTTPException(status_code=404, detail="Truck not found")
+
+    repair_history_list = truck.get("repair_history_list", [])
+    repair_history_list.append(repair.dict())
+    
+    update_result = collection.update_one(
+        {"_id": ObjectId(truck_id)},
+        {"$set": {"repair_history_list": repair_history_list}}
+    )
+    if update_result.modified_count == 1:
+        if (existing_truck := collection.find_one({"_id": ObjectId(truck_id)})) is not None:
+            return serialize_collection(existing_truck)
+    
+    raise HTTPException(status_code=500, detail="Failed to update repairs")
+
+@app.delete("/trucks/{truck_id}", response_description="Delete a truck")
+async def delete_truck(truck_id: str):
+    
+    collection = db.get_collection("trucks")
+    
+    delete_result = collection.delete_one({"_id": ObjectId(truck_id)})
+    if delete_result.deleted_count == 1:
+        return {"response": "Truck Deleted 204"}
+
+    return({"status_code": 404, "detail": f"Truck {truck_id} not found"})
+
+
+#Trailer API paths
+
+@app.get("/trailers/")
+async def get_trailers():
+    try:
+        collection = db.get_collection("trailers")
+        trailers = list(collection.find())
+        
+        return {"serialized_trailers": serialize_collection(trailers)}
+    except Exception as e:
+        print("An error occurred:", str(e))
+
+
+@app.get("/trailers/{trailer_id}")
+async def get_trailer(
+    trailer_id: Annotated[str, Path(title= "id of object to get")]
+):
+    try:
+        collection = db.get_collection("trailers")
+        trailer = collection.find_one({"_id": ObjectId(trailer_id)})
+        
+        return {"serialized_trailer": serialize_collection(trailer)}
+    except Exception as e:
+        print("An error occurred:", str(e))
+
+@app.post("/trailers/", response_description="Add new trailer", response_model=Trailer)
+async def create_trailer(trailer: Trailer = Body(...)):
+    try:
+        trailer = jsonable_encoder(trailer)
+        if "_id" in trailer:
+            del trailer["_id"]
+
+        collection = db.get_collection("trailers")
+
+        new_trailer = collection.insert_one(trailer)
+        created_trailer = collection.find_one({"_id": ObjectId(new_trailer.inserted_id)})
+        serialized_trailer = serialize_collection(created_trailer)
+        print({"serialized_trailer": serialized_trailer})
+        return serialized_trailer
+    except Exception as e:
+        print("An error occurred:", str(e))
+
```

Comment:
When splitting the logic in minor files, do the following:

Create a folder named: helpers

Each file should be called: service_name_helpers.py

And in those files you should have all of the logic for the helper. And then import in the controller that helper function and use it